interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,764 @@
+            f"Got '{sort_species}'"
+        )
+
+    all_elements = [ct.Element(e) for e in solution.element_names]
+    if sort_elements == "alphabetical":
+        all_elements = sorted(all_elements, key=lambda e: e.symbol)
+    elif sort_elements == "atomic-mass":
+        all_elements = sorted(all_elements, key=lambda e: e.weight)
+    elif sort_elements is not None:
+        raise ValueError(
+            "sort_elements must be None, 'alphabetical', or 'molar-mass'. "
+            f"Got '{sort_elements}'"
+        )
+
+    # Write output files
+    header_wrapper = HeaderTextWrapper(
+        [solution_name.name],
+        width=120,
+        initial_indent="! ",
+        subsequent_indent="! ",
+        break_long_words=False,
+        break_on_hyphens=False,
+    )
+    if "description" in solution.input_header:
+        header_text = (
+            header_wrapper.fill(
+                solution.input_header["description"],
+            )
+            + "\n"
+        )
+    else:
+        header_text = header_wrapper.fill("") + "\n"
+    mechanism_text = [
+        header_text,
+        build_elements_text(
+            all_elements,
+        ),
+        build_species_text(
+            all_species,
+        ),
+    ]
+    if thermo_path is None:
+        mechanism_text.append(
+            build_thermodynamics_text(
+                all_species,
+                separate_file=False,
+            ),
+        )
+    else:
+        thermo_text = [
+            header_text,
+            build_thermodynamics_text(
+                all_species,
+                separate_file=True,
+            ),
+        ]
+        thermo_path.write_text("\n".join(thermo_text))
+
+    # TODO: Handle phases without reactions
+    all_reactions = solution.reactions()
+    mechanism_text.append(build_reactions_text(all_reactions))
+
+    if transport_path is None and transport_exists:
+        mechanism_text.append(build_transport_text(all_species, separate_file=False))
+    elif transport_path is not None and transport_exists:
+        transport_text = [
+            header_text,
+            build_transport_text(all_species, separate_file=True),
+        ]
+        transport_path.write_text("\n".join(transport_text))
+
+    mechanism_path.write_text("\n".join(mechanism_text))
+
+    return output_files
+
+
+def main():
+    """
+    Parse command line arguments and pass them to `convert`
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Convert Cantera YAML input files to Chemkin-format mechanisms",
+    )
+
+    parser.add_argument("input", help="The input YAML filename. Required.")
+    parser.add_argument(
+        "--phase-name",
+        help=(
+            "Name of the phase to load from the input. If unspecified, the first "
+            "phase listed in the input file will be loaded."
+        ),
+        metavar="PHASE",
+        default="",
+    )
+
+    parser.add_argument(
+        "--mechanism",
```

Comment:
I believe `--output` would be more intuitive here.

Reply:
`--mechanism` is I think the typical name for the file, plus with (possibly) three separate file outputs, I don't think that `--output` would capture that. I think `--output` could make sense as the folder output option, but I'm concerned about adding too many options...

Reply:
Not a big issue …